### ISSUE TITLE
Allow `_` for readability in unitized constants like `3_day` or `5_gb`

### ISSUE THREAD

**Issue by bobhy** [nushell/nushell#9111]
### Related problem

```
〉100_000
100000
〉3.141_596_2828
3.1415962828
```

But not:
```
〉3day
3day
〉3_day
Error: nu::shell::external_command

  × External command failed
   ╭─[entry #14:1:1]
 1 │ 3_day
   · ──┬──
   ·   ╰── executable was not found
   ╰────
  help: No such file or directory (os error 2)
```

### Describe the solution you'd like

So one could say:
```
let duration = 3.165_hours
let size = 1.234_567__mb
```
Using `_` for improving readability of numeric constants is a great idea.  Why not extend it to unitized constants of all types?  Duration, Filesize, (what's the complete list of these?)

**Comment by fdncred** [comment]
I'd support this, only if it was in addition to what we have. I like what you said in another issue that there should be multiple ways to express this. e.g.
- 3day
- 3days
- 3_days
- 3da
- 3_da
etc

**Comment by bobhy** [comment]
Yes, it could be a pure add.  
This is a good first project for an aspiring parser hacker, documented here for anyone who wants to take a crack at it.

The routines that parse [Duration](https://github.com/nushell/nushell/blob/6dc7ff2335c9547182094451cd40c19e66c0678b/crates/nu-parser/src/parser.rs#L2250-L2329) and [Filesize](https://github.com/nushell/nushell/blob/6dc7ff2335c9547182094451cd40c19e66c0678b/crates/nu-parser/src/parser.rs#L2351-L2429) are pretty straightforward.  Each has a list of suffixes and you could add more easily. 
If you only worry about underscores between the numeric and suffix part, each has just one place you'd have to do that scan. (fully general placement would allow underscores in the numeric part too, Nu allows this for ints and float literals now.)
The hardest part would be adding enough test cases for the new syntax.  I recommend `rstest`, it's already in the project.

**Comment by bobhy** [comment]
Just to make this an even more inviting first parser project for someone...
The above would allow duration or filesize literals you type into Nu to include underscores.

To have the duration print with an underscore, see [TimePeriod::to_text()](https://github.com/nushell/nushell/blob/0ea973b78b57bb01ae0b58ecd1c02fda21184043/crates/nu-protocol/src/value/mod.rs#L3427-L3442).  If you add underscores before these duration units, they will print out just fine.

Filesize is a bit of a puzzle, I don't have any suggestions for it yet.

**Comment by fdncred** [comment]
It would be also be nice if plurals could be used when appropriate.
